mlir/lib/Analysis/DataFlow/SparseAnalysis.cpp

```diff
@@ -609,12 +609,24 @@ void AbstractSparseBackwardDataFlowAnalysis::visitRegionSuccessors(
            *getLatticeElementFor(getProgramPointAfter(op), input));
       unaccounted.reset(operand.getOperandNumber());
     }
+
+    if (successor.isParent())
+      continue;
+    auto arguments = successor.getSuccessor()->getArguments();
+    for (BlockArgument argument : arguments) {
+      if (llvm::find(inputs, argument) == inputs.end()) {
+        regionArguments.push_back(argument);
+      }
+    }
   }
   // All operands not forwarded to regions are typically parameters of the
   // branch operation itself (for example the boolean for if/else).
   for (int index : unaccounted.set_bits()) {
     visitBranchOperand(op->getOpOperand(index));
   }
+  for (BlockArgument argument : regionArguments) {
+    visitBranchRegionArgument(argument);
```

Comment:
It looks like we are calling `visitBranchRegionArgument` only for block arguments that not "control flow block arguments". E.g., for an `scf.for`, this would be called for the IV but not for the loop-carried variables. Is that correct? And why is that the case?

Assuming that's correct, should this function maybe be called `visitNonControlFlowArguments`, similar to the `SparseForwardDataFlowAnalysis`?